bento_beacon/utils/gohan_utils.py

```diff
@@ -126,17 +128,39 @@ def range_query_to_gohan(beacon_args, granularity, ids_only):
 
 
 def bracket_query_to_gohan(beacon_args, granularity, ids_only):
-    print("BRACKET QUERY")
+    current_app.logger.debug("BRACKET QUERY")
     # TODO
     # either implement here by filtering full results, or implement directly in gohan
     raise NotImplemented(message="variant bracket query not implemented")
 
 
 def geneId_query_to_gohan(beacon_args, granularity, ids_only):
-    print("GENE ID QUERY")
-    # TODO
-    # determine assembly, call gohan for gene coordinates, launch search
-    raise NotImplemented(message="variant geneId query not implemented")
+    current_app.logger.debug("GENE ID QUERY")
+    gene_id = beacon_args.get("geneId")
+    assembly_from_query = beacon_args.get("assemblyId")
+
+    # query all assemblies present in gohan if not specified
+    assemblies = [assembly_from_query] if assembly_from_query is not None else gohan_assemblies()
+    gohan_args = beacon_to_gohan_generic_mapping(beacon_args)
+
+    gohan_results = []
+    for assembly in assemblies:
```

Comment:
make a comment here indicating that this has to be redone with async, since it's doing sequential HTTP requests.

Reply:
yeah, I will do all the async together rather than piecemeal. Will async even do anything here? Repeated calls to the same two services?

Reply:
since reference is async, it theoretically lets it execute two database calls "at once". not parallelized, but passing both requests to postgres and waiting for them concurrently which [should...] give some speed up in some situations, since the postgres connection can do parallel reads/searchers.

Reply:
specifically by using async concurrency here i mean with https://docs.python.org/3/library/asyncio-task.html#asyncio.gather